File: php/features.php
<?php
declare(strict_types=1);

// HostedRest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class HostedRestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HostedRestBaseFeature();
            case "test":
                return new HostedRestTestFeature();
            default:
                return new HostedRestBaseFeature();
        }
    }
}

File: rb/features.rb
# HostedRest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module HostedRestFeatures
  def self.make_feature(name)
    case name
    when "base"
      HostedRestBaseFeature.new
    when "test"
      HostedRestTestFeature.new
    else
      HostedRestBaseFeature.new
    end
  end
end
